src/App.tsx

```diff
@@ -1,39 +1,48 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
 
+import { Clock } from './Clock';
+
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+  componentDidMount() {
+    document.addEventListener('contextmenu', () => this.setState(
+      { hasClock: false },
+    ));
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('click', () => this.setState(
+      { hasClock: true },
+    ));
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    window.setInterval(
+      () => this.setState({ clockName: getRandomName() }), 3300,
+    );
```

Comment:
You should clear this interval on componentWillUnmount in cases, when App component can be destroyed